test/python/compiler/test_compiler.py

```diff
@@ -171,6 +171,48 @@ def test_transpiler_layout_from_intlist(self):
 
         self.assertEqual(mapped_qubits, [4, 6, 10])
 
+    def test_wrong_layout_in_single_qubit_circuits(self):
+        """ If the layout is wrong (even in single qubit circuits), raise.
+            See https://github.com/Qiskit/qiskit-terra/issues/2038
+        """
+        qr = QuantumRegister(1, 'qr')
+        qc = QuantumCircuit(qr)
+        qc.h(qr[0])
+
+        layout = {(qr, 0): 19}  # 19 is not the coupling map
+
+        cmap = [[1, 0], [1, 2], [2, 3], [4, 3], [4, 10],
+                [5, 4], [5, 6], [5, 9], [6, 8], [7, 8],
+                [9, 8], [9, 10], [11, 3], [11, 10],
+                [11, 12], [12, 2], [13, 1], [13, 12]]
+
+        self.assertRaises(MapperError, transpile, qc,
+                          backend=None,
+                          coupling_map=cmap,
+                          basis_gates=['h'],
+                          initial_layout=layout)
+
+    def test_transpile_single_qubit(self):
+        """ A single-qubit circuit transpilation
+        """
+        qr = QuantumRegister(1, 'qr')
+        qc = QuantumCircuit(qr)
+        qc.h(qr[0])
+        layout = {(qr, 0): 12}
+        cmap = [[1, 0], [1, 2], [2, 3], [4, 3], [4, 10],
+                [5, 4], [5, 6], [5, 9], [6, 8], [7, 8],
+                [9, 8], [9, 10], [11, 3], [11, 10],
+                [11, 12], [12, 2], [13, 1], [13, 12]]
+
+        result = transpile(qc, backend=None, coupling_map=cmap, basis_gates=['u2'],
+                           initial_layout=layout)
+
+        qr = QuantumRegister(14, 'q')
+        expected = QuantumCircuit(qr)
+        expected.u2(0, 3.1416, qr[12])
```

Comment:
can you actually go to Qobj here (use `qiskit.compiler.assemble_circuits`), and check that the QobjInstruction  for`u2` is applied to physical qubit 12? That will be the most reassuring means by which we can say layout was done correctly.

Reply:
done in f6db1d28